File: aemet/Aemet-Java-IvanRC/src/main/java/dto/MedicionDto.java
package dto;

import com.squareup.moshi.Json;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Json(name = "medicion")
@Root(name = "medicion")
public class MedicionDto {
    @Element(name = "poblacion", required = false)
    String poblacion = null;

    @Element(name = "provincia", required = false)
    String provincia = null;

    @Element(name = "temperatura_max", required = false)
    String temperaturaMax = null;

    @Element(name = "hora_temperatura_max", required = false)
    String horaTempMax = null;

    @Element(name = "temperatura_min", required = false)
    String temperaturaMin = null;

    @Element(name = "hora_temperatura_min", required = false)
    String horaTempMin = null;

    @Element(name = "precipitacion", required = false)
    String precipitacion = null;

    @Element(name = "dia", required = false)
    String dia = null;
}

File: aemet/Aemet-kotlin-IvanRC/src/main/kotlin/dto/MedicionDto.kt
package dto

import com.squareup.moshi.Json
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Json(name = "medicion")
@Root(name = "medicion")
data class MedicionDto (
    @field:Element(name = "poblacion")
    @param:Element(name = "poblacion")
    val poblacion: String,

    @field:Element(name = "provincia")
    @param:Element(name = "provincia")
    val provincia: String,

    @field:Element(name = "temperatura_maxima")
    @param:Element(name = "temperatura_maxima")
    val temperaturaMax: String,

    @field:Element(name = "hora_de_temperatura_maxima")
    @param:Element(name = "hora_de_temperatura_maxima")
    val horaTempMax: String,

    @field:Element(name = "temperatura_minima")
    @param:Element(name = "temperatura_minima")
    val temperaturaMin: String,

    @field:Element(name = "hora_de_temperatura_minima")
    @param:Element(name = "hora_de_temperatura_minima")
    val horaTempMin: String,

    @field:Element(name = "precipitacion")
    @param:Element(name = "precipitacion")
    val precipitacion: String,

    @field:Element(name = "dia")
    @param:Element(name = "dia")
    val dia: String
)